engine/apps/api/permissions.py

```diff
@@ -250,6 +247,17 @@ class Permissions:
             Resources.OTHER_SETTINGS, Actions.WRITE, LegacyAccessControlRole.ADMIN
         )
 
+        # NOTE: we don't currently add the label delete permission here because we don't currently use this in OnCall
+        LABEL_CREATE = LegacyAccessControlCompatiblePermission(
+            Resources.LABEL, Actions.CREATE, LegacyAccessControlRole.EDITOR, prefix=PluginID.LABELS
+        )
```

Comment:
I only see the `Labels Reader` and `Labels Writer` roles in the existing Grafana RBAC UI, is this PR introducing a new one for create? maybe it's a good idea to only keep read and write (so create is part of write).

Reply:
tldr; no it's not introducing any new RBAC roles. The `LABEL_CREATE`, `LABEL_READ`, and `LABEL_WRITE` variables are just helpers which translate to the permission strings `grafana-labels-app:label:create`, `grafana-labels-app:label:read`, and `grafana-labels-app:label:write`, respectively.

---

Those are the RBAC _roles_ that're available from Labels. Roles grant >= 1 _permission(s)_ . [These](https://github.com/grafana/gops-labels/blob/main/plugin/src/plugin.json#L30-L54) are the permissions granted by those labels RBAC roles:
```json
"roles": [
    {
      "role": {
        "name": "Labels Writer",
        "description": "Read/write/create/delete Labels",
        "permissions": [
          { "action": "plugins.app:access", "scope": "plugins:id:grafana-labels-app" },
          { "action": "grafana-labels-app.label:read" },
          { "action": "grafana-labels-app.label:write" },
          { "action": "grafana-labels-app.label:create" },
          { "action": "grafana-labels-app.label:delete" }
        ]
      },
      "grants": ["Admin", "Editor"]
    },
    {
      "role": {
        "name": "Labels Reader",
        "description": "Read-only access to Labels",
        "permissions": [
          { "action": "plugins.app:access", "scope": "plugins:id:grafana-labels-app" },
          { "action": "grafana-labels-app.label:read" }
        ]
      },
      "grants": ["Viewer"]
    }
  ]
```

so in OnCall we're basically validating that the user has the necessary _permission_ (action), we don't necessarily care about the role names themselves.

I agree that `:write` and `:create` should be consolidated over in labels but that's the current state of things. If you're curious, see these resources for more context:
- https://github.com/grafana/gops-labels/blob/main/pkg/plugin/auth.go#L42
- https://github.com/grafana/gops-labels/blob/main/pkg/plugin/plugin.go#L70-L74
- https://github.com/grafana/gops-labels/blob/main/pkg/plugin/plugin.go#L84-L96